app/models/emoji.rb

```diff
@@ -239,6 +239,13 @@ def self.gsub_emoji_to_unicode(str)
   end
 
   def self.lookup_unicode(name)
+    # ignore denied emojis and their aliases
+    if SiteSetting.emoji_deny_list.present?
+      denied_emoji = SiteSetting.emoji_deny_list.split("|")
+      denied_emoji += denied_emoji.map { |e| Emoji.aliases[e] }.flatten.compact
```

Comment:
```suggestion
      denied_emoji.concat(denied_emoji.flat_map { |e| Emoji.aliases[e] }.compact)
```

Minor suggestion that with less allocations.